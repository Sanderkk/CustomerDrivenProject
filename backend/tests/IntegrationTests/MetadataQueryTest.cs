using HotChocolate;
using HotChocolate.Execution;
using HotChocolate.Types;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using src.Api.Inputs;
using src.Api.Queries;
using src.Config;
using src.Database;
using src.Api.Types;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using Xunit;

namespace tests.IntegrationTests
{
    public class MetadataQueryTest
    {
        //Database Connect config (move to own config file)
        private string dbConnectionString = "Host=sanderkk.com;Username=sintef;Password=123456;Database=fishfarm2";

        [Fact]
        public async Task AllMetadataQuery()
        {
            //Arrange
            //Connect to database
            IServiceProvider serviceProvider =
                new ServiceCollection()
                .AddSingleton<IFishFarmRepository, FishFarmRepository>()
                .AddSingleton<IDatabaseConfig>(sp =>
                new DatabaseConfig()
                {
                    DatabaseConnectionString = dbConnectionString
                }
                )
                .BuildServiceProvider();

            IQueryExecutor executor = Schema.Create(c =>
            {
                c.RegisterQueryType(new ObjectType<MetadataQuery>(d => d.Name("Query")));
                c.RegisterType<List<MetadataType>>();
            })
            .MakeExecutable();

            IReadOnlyQueryRequest request =
                QueryRequestBuilder.New()
                .SetQuery(@"{allMetadata(sensorID:1){metadataID,sensorID,name,serialNumber,number,department,company,purchaseDate,lending,lendingPrice,cableLength,checkOnInspectionRound,company,voltage,coordinate, altitude, locationDescription, createdAt, updatedAt, outdatedFrom}}")
                .SetServices(serviceProvider)
                .Create();

            // act
            IExecutionResult result = await executor.ExecuteAsync(request);
            //Assert
            Snapshot.Match(result);
        }

        [Fact]
        public async Task LastMetadataQuery()
        {
            //Arrange
            //Connect to database
            IServiceProvider serviceProvider =
                new ServiceCollection()
                .AddSingleton<IFishFarmRepository, FishFarmRepository>()
                .AddSingleton<IDatabaseConfig>(sp =>
                new DatabaseConfig()
                {
                    DatabaseConnectionString = dbConnectionString
                }
                )
                .BuildServiceProvider();

            IQueryExecutor executor = Schema.Create(c =>
            {
                c.RegisterQueryType(new ObjectType<MetadataQuery>(d => d.Name("Query")));
                c.RegisterType<List<MetadataType>>();
            })
            .MakeExecutable();

            IReadOnlyQueryRequest request =
                QueryRequestBuilder.New()
                .SetQuery(@"{lastMetadata(sensorID:1){metadataID,sensorID,name,serialNumber,number,department,company,purchaseDate,lending,lendingPrice,cableLength,checkOnInspectionRound,company,voltage,coordinate, altitude, locationDescription, createdAt, updatedAt, outdatedFrom}}")
                .SetServices(serviceProvider)
                .Create();

            // act
            IExecutionResult result = await executor.ExecuteAsync(request);
            //Assert
            Snapshot.Match(result);
            //result.MatchSnapshot();
        }
    }
}