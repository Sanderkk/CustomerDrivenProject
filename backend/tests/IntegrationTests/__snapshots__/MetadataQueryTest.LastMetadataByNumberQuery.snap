{
  "Data": {
    "metadata": [
      {
        "metadataID": 194,
        "sensorID": 1,
        "locationID": 2,
        "name": "Oxygen optode 4531D",
        "serialNumber": "588",
        "number": "RO.038-50m",
        "modelNumber": "4531D",
        "department": null,
        "company": null,
        "ownerID": null,
        "purchaseDate": null,
        "lending": null,
        "lendingPrice": null,
        "cableLength": 50.0,
        "checkOnInspectionRound": null,
        "inspectionRound": null,
        "servicePartner": null,
        "voltage": "12V DC",
        "coordinate": "55°24'12.2″N 2°10'26.5″E",
        "altitude": 500,
        "locationDescription": "nowimhere",
        "identificator": null,
        "measureArea": null,
        "picture": null,
        "plannedDisposal": null,
        "actualDisposal": null,
        "warrantyDate": null,
        "signal": "RS232",
        "tag1": null,
        "tag2": null,
        "tag3": null,
        "timeless": null,
        "tollerance": null,
        "createdAt": "2020-10-19T12:11:38.000+02:00",
        "updatedAt": "2020-10-19T12:11:38.000+02:00",
        "outdatedFrom": null
      }
    ]
  },
  "Extensions": {},
  "Errors": [],
  "ContextData": {}
}
